keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -293,7 +293,14 @@ def _augment(self, inputs):
         return result
 
     def _batch_augment(self, inputs):
-        return self._map_fn(self._augment, inputs)
+        if BOUNDING_BOXES in inputs:
+            inputs[BOUNDING_BOXES] = self._pad_bounding_boxes(inputs[BOUNDING_BOXES])
+        augmented_batch = self._map_fn(self._augment, inputs)
+        if BOUNDING_BOXES in inputs:
+            augmented_batch[BOUNDING_BOXES] = self._filter_padded_boxes(
+                augmented_batch[BOUNDING_BOXES]
+            )
+        return augmented_batch
```

Comment:
I think what we should do is add a new return value to _format_inputs that stores whether or not bounding boxes were ragged when passed in.

Let's convert the return type of `_format_inputs()` to be two dictionaries: the inputs dict, and a metadata dict containing "if inputs are single image", "if bboxes were ragged', etc.

Reply:
Done